libnavigator/src/test/java/com/mapbox/navigation/navigator/NavigationOkHttpServiceTest.kt

```diff
@@ -0,0 +1,231 @@
+package com.mapbox.navigation.navigator
+
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.common.HttpMethod
+import com.mapbox.common.HttpRequest
+import com.mapbox.common.HttpResponse
+import com.mapbox.common.HttpResponseCallback
+import com.mapbox.common.UserAgentComponents
+import com.mapbox.navigation.navigator.NavigationOkHttpService.Companion.GZIP
+import com.mapbox.navigation.navigator.NavigationOkHttpService.Companion.HEADER_ENCODING
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import java.io.IOException
+import java.util.Locale
+import okhttp3.Call
+import okhttp3.Dispatcher
+import okhttp3.Headers
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.ResponseBody
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+
+class NavigationOkHttpServiceTest {
+    private val logger: Logger = mockk(relaxUnitFun = true)
+    private val httpClientBuilder: OkHttpClient.Builder = mockk(relaxUnitFun = true)
+    private val httpClient: OkHttpClient = mockk(relaxUnitFun = true)
+    private val httpDispatcher: Dispatcher = mockk(relaxUnitFun = true)
+
+    private val nativeRequest = HttpRequest(
+        HttpMethod.GET,
+        "https://api.mapbox.com/xyz",
+        hashMapOf("User-Agent" to "MapboxNavigationNative/X.Y.Z MapboxNavigationNative"),
+        true,
+        UserAgentComponents.Builder().build()
+    )
+    private val nativeCallback: HttpResponseCallback = mockk(relaxUnitFun = true)
+
+    private lateinit var httpService: NavigationOkHttpService
+
+    @Before
+    fun setup() {
+        every { httpClientBuilder.build() } returns httpClient
+        every { httpClientBuilder.addInterceptor(any()) } returns httpClientBuilder
+        every { httpClient.dispatcher() } returns httpDispatcher
+        httpService = NavigationOkHttpService(httpClientBuilder, logger)
+    }
+
+    @Test
+    fun setMaxRequestsPerHost() {
+        httpService.setMaxRequestsPerHost(5)
+        verify { httpDispatcher.maxRequestsPerHost = 5 }
```

Comment:
NIT Normally is good to structure the tests following the triple A 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=40 http://wiki.c2.com/?ArrangeActAssert
If the `A`rrange part gets too long we should create factory or builder methods.

Also I'm not a big fan of inline setup in tests because among other things it hides the Arrange part completely.